# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentChild.h
# Commit: aea23435804b
# Full Hash: aea23435804b56798dbae8fa46dfccb8cbdca131
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2019-12-13 16:53:24
# Regressor Bug: 1593246
# File Overlap Count: 6
# Description:
#   Bug 1593246 - Part 5: Replicate session storage data r=janv,asuth
#   
#   This commit implements a simple data replication scheme, where
#   each content process receives a copy of the relevant session
#   storage data when navigating, and sending all its session storage
# ==============================================================================

diff -r da1672f831d1 -r aea23435804b dom/ipc/ContentChild.h
--- a/dom/ipc/ContentChild.h	Fri Dec 13 09:16:32 2019 +0000
+++ b/dom/ipc/ContentChild.h	Fri Dec 13 09:45:55 2019 +0000
@@ -693,6 +693,11 @@
   mozilla::ipc::IPCResult RecvEvictContentViewers(
       nsTArray<uint64_t>&& aToEvictSharedStateIDs);
 
+  mozilla::ipc::IPCResult RecvSessionStorageData(
+      BrowsingContext* aTop, const nsACString& aOriginAttrs,
+      const nsACString& aOriginKey, const nsTArray<KeyValuePair>& aDefaultData,
+      const nsTArray<KeyValuePair>& aSessionData);
+
 #ifdef NIGHTLY_BUILD
   // Fetch the current number of pending input events.
   //